job_executor/app.py

```diff
@@ -229,11 +243,22 @@ def main():
                     f" (worker, built, queued manager jobs)"
                 )
             for job in queued_worker_jobs:
-                if len(workers) < NUMBER_OF_WORKERS:
+                job_size = local_storage.get_size_in_bytes(job.dataset_name)
+                if job_size == 0:
+                    logger.info(
+                        f"{job.job_id} Failed to get the size of the dataset."
+                    )
+                    raise LocalStorageError(
```

Comment:
If we do it like this the local storage module itself might as well raise the error so we don't need to do the check here.

Another option would be to fail the job:
```py
        job_service.update_job_status(
            job_id, "failed", log=f"No such dataset available for import"
        )
```
This is preferable to crashing the whole executor if one dataset should be tampered with in some unexpected way.